### PR TITLE
French phonetic filter

### PR DESCRIPTION
**Pull request by hcapitaine**
French phonetic filter based on French pronunciation.

### CODE REVIEW

File: plugins/analysis-phonetic/src/main/java/org/elasticsearch/index/analysis/phonetic/FrenchPhonetic.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+ * </pre>
+ *
+ * @see <a href="https://fr.wikipedia.org/wiki/Prononciation_du_fran%C3%A7ais#Prononciation_des_graph.C3.A8mes"/> French prononciation on Wikipedia</a>
+ *
+ * @author Harold Capitaine
+ * @author Galeries Lafayette
+ */
+public class FrenchPhonetic implements StringEncoder {
+
+    private static final List<Character> VOWELS = Arrays.asList('A', 'E', '2', 'I', 'O', 'U', 'Y');
```
Comment (dadoonet) [inline plugins/analysis-phonetic/src/main/java/org/elasticsearch/index/analysis/phonetic/FrenchPhonetic.java]: Why `2` here?

**Comment by dadoonet** [comment]
Hi @hcapitaine! 

Could you sign the CLA?

I'm also wondering if this should be contributed to Lucene? @jpountz WDYT?

**Comment by hcapitaine** [comment]
Hi David,

I already signed the cla and even received a receipt but it didn't
correctly worked after that.

File: plugins/analysis-phonetic/src/main/java/org/elasticsearch/index/analysis/phonetic/FrenchPhonetic.java (hunk 1 above)
Comment (hcapitaine) [inline plugins/analysis-phonetic/src/main/java/org/elasticsearch/index/analysis/phonetic/FrenchPhonetic.java]: 2 is a french sound made only of vowels and for example baisé and baissé
would have been equally translated as B2S2 but not as B2Z2 and B2S2 if 2
was not included in the constant.

Worked mainly with all case VOWELS constant is used in.

If you find a non working case I will put the 2 in if statements where
needed and remove it from the constant.

Contribute to the tests ;)

**Comment by karmi** [comment]
@hcapitaine Seems like you have signed with a different e-mail then is used in your Git commit (and possibly on record in your Github profile). If you add your Gmail e-mail address to Github, it should work next time we check the records -- that's the easiest way. Ping me if you need more help!

**Comment by hcapitaine** [comment]
Ok you're totally right. I added the email I commit with in my github
account. It should be ok now.

**Comment by karmi** [comment]
@hcapitaine For some reason the check still fails, even when I re-trigger all the checks... We'll try to look into it! But we have received your signature properly, thanks!

@dadoonet Can you process the pull request? I think we can consider Harold as properly signed up for CLA.

**Comment by ypereirareis** [comment]
Hi @hcapitaine !

How to use your work/plugin right now with elasticsearch ?

Thx :smile:

**Comment by karmi** [comment]
Hi @hcapitaine, I've tried to re-trigger a check for this particular PR to better debug this situation, but it still fails. Are you sure you've added both of your e-mails (gmail.com and galerieslafayette.com) to your Github profile? You've signed with your Gmail, and commited with your work mail. In theory, Github should match these e-mails with your Github profile, and it should work on our side. Can you please verify that both of these e-mails are attached to your Github profile?

**Comment by hcapitaine** [comment]
Hi @karmi, 

my galeries address is now added to my account. It should work right now.

**Comment by hcapitaine** [comment]
Hi @ypereirareis,

My work is not really tied to a specific version of elasticsearch except for the tests. Maybe I could do some others PR to include it in other versions of ES. @dadoonet what is your policy for backport?

**Comment by hcapitaine** [comment]
@ypereirareis 
If you want to use it right now, you can checkout specific version of ES that you want. Apply diffs except integration test part at the same path in the code and it should work. Compile and use the generated plugin in your ES cluster.

**Comment by karmi** [comment]
@hcapitaine Thanks!, I'm super-glad that it works as it should -- ie. since the e-mail is attached to Github, we can match you to the signature, and your comment triggers the check again.

**Comment by clintongormley** [comment]
@rmuir would you mind taking a look at this?  Should this be submitted to Lucene instead?

**Comment by jpountz** [comment]
We just discussed it in Fixit Friday. The analysis algorithms that we expose are usually backed by standards (eg. for normalization) or research papers that try to prove the quality of the algorithm. Since this one does not appear to have one, we would rather like it be remain a community plugin, which could then be listed at https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis.html#_community_contributed_analysis_plugins (just open a PR that edits `docs/plugins/analysis.asciidoc` when you publish it).

**Pull request closed without merging**